### ISSUE TITLE
[FEATURE] - run a job on the last weekday of the month

### ISSUE THREAD

**Issue by rsvix** [go-co-op/gocron#804]
### Describe the bug

Gocron cronjob cant parse "LW", errors with invalid syntax

### To Reproduce

`
package main
import (
	"log"
	"github.com/go-co-op/gocron/v2"
)
func main() {
	s, err := gocron.NewScheduler()
	if err != nil {
		log.Fatal(err)
	}

	_, err = s.NewJob(
		gocron.CronJob("0 0 5 LW * ?", true),
		gocron.NewTask(func() {
			log.Println("Test job 1")
		}),
	)
	if err != nil {
		log.Printf("Test job 1 err: %v", err)
	}

	_, err = s.NewJob(
		gocron.CronJob("0 5 LW * ?", false),
		gocron.NewTask(func() {
			log.Println("Test job 2")
		}),
	)
	if err != nil {
		log.Printf("Test job 2 err: %v", err)
	}

	s.Start()
	for {
	}
}
`

### Version

v2.13.0

### Expected behavior

The expression should be parsed since its a valid cron expression

**Comment by JohnRoesler** [comment]
Hi @rsvix,

A few thoughts:
- `L W ?` are all consider non-standard, and aren't including in all Cron implementations. https://en.wikipedia.org/wiki/Cron#Non-standard_characters
- gocron uses https://github.com/robfig/cron for parsing cron syntax, and it's not supported over there, although there are some issues / PRs related
  - https://github.com/robfig/cron/issues/299
  - https://github.com/robfig/cron/issues/319
  - https://github.com/robfig/cron/pull/325

Is it fair to say, this is actually a feature request to support:
- The ability to run a job on the last weekday of the month?

**Comment by rsvix** [comment]
Hello @JohnRoesler,

Thats fair, i will update it to a feature request
Sad that https://github.com/robfig/cron seems to be abandoned and the PRs are still not merged.

**Comment by Dojeto** [comment]
I would like you suggest you to change the parser lib which you are currently using  

https://github.com/adhocore/gronx this lib can help you to make a new feature to this lib

https://github.com/adhocore/gronx?tab=readme-ov-file#modifiers

**Comment by JohnRoesler** [comment]
Thanks @Dojeto I'll take a look into that library!

**Comment by Dojeto** [comment]
I would love to contribute to this if you can help me with the testing, test cases are keep getting failed even on 

--- FAIL: TestScheduler_WithMonitor (0.59s)
    scheduler_test.go:39: found unexpected goroutines:

**Comment by JohnRoesler** [comment]
@Dojeto contribution would be appreciated! For the tests, you can set the env TEST_ENV=ci. I've been debugging these leaks for a while. You can ignore for now

**Comment by Dojeto** [comment]
@JohnRoesler i am trying to solve this issue but my test cases are getting failed but i am not clear at which point it is getting is it something related to resource leak ? i am not familiar with the go testing tool, it would be great if you could help me with this :)

![Image](https://github.com/user-attachments/assets/2672be2d-1e07-409f-bdc6-87c6ce33bce2)

**Comment by JohnRoesler** [comment]
@Dojeto try pulling in the v2 branch again, the resource leaks should be fixed now.

**Comment by Dojeto** [comment]
@JohnRoesler i am trying to implement this feature using this https://github.com/adhocore/gronx lib and since there are lot of things different here that might not gives the backward compatibility, since this package is well maintained i would highly suggest you to migrate

These are the following imacts of the changing package : 

- since these package automatically detect the cron field there no need to explicitly say withSeconds `true` 
- error will change since online one validation function is required to validate crons

even after using that lib LW is non standard we have to write own logic for that 😄

**Comment by JohnRoesler** [comment]
Hey @Dojeto 

A few things here - I think we should make an interface for the cron implementation. Then, if tomorrow there is a new cron library that supports LW (and or any other non-standard cron syntax someone wants) they can implement the interface and pass it into gocron.

As for the specific issues, I don't want to change the gocron package, so we'll need to take that into account with the interface and then the implementations with robfig/cron and adhocore/gronx. I'd likely keep robfig/cron as the default, but allow for users to opt to a different implementation if desired.

**Comment by Dojeto** [comment]
@JohnRoesler that's really great idea, also users can write own non standard syntax i would love to contribute to this so what's your plan this implementation for the next version ( v3 ) ? 

if you are planning for the v3 then just create a todo list so others can contribute.

**Comment by JohnRoesler** [comment]
I think this can be done in v2 without breaking anything.

1. Abstract the current cron implementation behind an interface.
2. Add a new JobOption (i'd make it a job option so that it can be done for a single job or globally) that takes in the cron interface. If that method is never called, use the default.

**Comment by Dojeto** [comment]
`func CronJob(crontab string, withSeconds bool) JobDefinition {
	return cronJobDefinition{
		crontab:     crontab,
		withSeconds: withSeconds,
	}
}`

current Cronjob function takes extra field `withSeconds` there is not significance of this if we use our own custom cron cause `IsValid()` function takes care of it, we can remove it and define own logic in setup like user passed the 6 field or 5 we can decide and parse (default : robfig/cron) accordingly, but if we remove it then it is breaking changes 

We can also make new `CustomCron()` which only takes the cron string.

**Comment by JohnRoesler** [comment]
Yeah, don't remove, even if some implementations of the interface won't require the bool withSeconds.
